commit 53dabce2652fb854eae84609ce9c37429d5d87ba
Author: Vlastimil Babka <vbabka@suse.cz>
Date:   Thu Jul 11 18:35:31 2024 +0200

    mm, page_alloc: put should_fail_alloc_page() back behing CONFIG_FAIL_PAGE_ALLOC
    
    This mostly reverts commit af3b854492f3 ("mm/page_alloc.c: allow error
    injection").  The commit made should_fail_alloc_page() a noinline function
    that's always called from the page allocation hotpath, even if it's empty
    because CONFIG_FAIL_PAGE_ALLOC is not enabled, and there is no option to
    disable it and prevent the associated function call overhead.
    
    As with the preceding patch "mm, slab: put should_failslab back behind
    CONFIG_SHOULD_FAILSLAB" and for the same reasons, put the
    should_fail_alloc_page() back behind the config option.  When enabled, the
    ALLOW_ERROR_INJECTION and BTF_ID records are preserved so it's not a
    complete revert.
    
    Link: https://lkml.kernel.org/r/20240711-b4-fault-injection-reverts-v1-2-9e2651945d68@suse.cz
    Signed-off-by: Vlastimil Babka <vbabka@suse.cz>
    Cc: Akinobu Mita <akinobu.mita@gmail.com>
    Cc: Alexei Starovoitov <ast@kernel.org>
    Cc: Andrii Nakryiko <andrii@kernel.org>
    Cc: Christoph Lameter <cl@linux.com>
    Cc: Daniel Borkmann <daniel@iogearbox.net>
    Cc: David Rientjes <rientjes@google.com>
    Cc: Eduard Zingerman <eddyz87@gmail.com>
    Cc: Hao Luo <haoluo@google.com>
    Cc: Hyeonggon Yoo <42.hyeyoo@gmail.com>
    Cc: Jiri Olsa <jolsa@kernel.org>
    Cc: John Fastabend <john.fastabend@gmail.com>
    Cc: KP Singh <kpsingh@kernel.org>
    Cc: Martin KaFai Lau <martin.lau@linux.dev>
    Cc: Mateusz Guzik <mjguzik@gmail.com>
    Cc: Roman Gushchin <roman.gushchin@linux.dev>
    Cc: Song Liu <song@kernel.org>
    Cc: Stanislav Fomichev <sdf@fomichev.me>
    Cc: Yonghong Song <yonghong.song@linux.dev>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>

diff --git a/mm/fail_page_alloc.c b/mm/fail_page_alloc.c
index b1b09cce9394..532851ce5132 100644
--- a/mm/fail_page_alloc.c
+++ b/mm/fail_page_alloc.c
@@ -1,5 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0
 #include <linux/fault-inject.h>
+#include <linux/error-injection.h>
 #include <linux/mm.h>
 
 static struct {
@@ -21,7 +22,7 @@ static int __init setup_fail_page_alloc(char *str)
 }
 __setup("fail_page_alloc=", setup_fail_page_alloc);
 
-bool __should_fail_alloc_page(gfp_t gfp_mask, unsigned int order)
+bool should_fail_alloc_page(gfp_t gfp_mask, unsigned int order)
 {
 	int flags = 0;
 
@@ -41,6 +42,7 @@ bool __should_fail_alloc_page(gfp_t gfp_mask, unsigned int order)
 
 	return should_fail_ex(&fail_page_alloc.attr, 1 << order, flags);
 }
+ALLOW_ERROR_INJECTION(should_fail_alloc_page, TRUE);
 
 #ifdef CONFIG_FAULT_INJECTION_DEBUG_FS